### DISCUSSION TITLE
No face crops at all - no permission or disk full

### DISCUSSION THREAD

**Discussion by monarc83** [photoprism/photoprism#2036] [Q&A] [answered]
I have installed PhotoPrism as a container on my Unraid Server using the photoprism/photoprism repository.

I have encountered the issue that I cannot see any faces (crops) for recognized faces at all. Not under People->New, not when I transferred them to People->Recognized and not under Edit->People for a specific photo. I can see below unexpected log message which is for sure the reason.

time="2022-02-16T15:55:48+01:00" level=error msg="failed saving 9f796230b5702640eeb48365d72bd1620cff4089_320x320_crop_1d00dd1591cc.jpg - no permission or disk full?"

But why I am getting this log message? From my point of view everything is looking fine. Any ideas?

df -h
Filesystem      Size  Used Avail Use% Mounted on
/dev/loop2       30G   17G   12G  59% /
tmpfs            64M     0   64M   0% /dev
tmpfs           6.9G     0  6.9G   0% /sys/fs/cgroup
shm              64M     0   64M   0% /dev/shm
/dev/nvme0n1p1  932G   74G  857G   8% /photoprism/storage
shfs            3.7T  640G  3.1T  18% /photoprism/originals
/dev/loop2       30G   17G   12G  59% /etc/hosts
tmpfs           6.9G     0  6.9G   0% /proc/acpi
tmpfs           6.9G     0  6.9G   0% /sys/firmware

ls -l
total 0
drwxrwxrwx 1 photoprism photoprism  94 Jan 21 12:50 assets
drwxrwxrwx 1 photoprism photoprism  20 Jan 21 12:50 bin
drwxrwxrwx 1 photoprism users        6 Feb 15 12:46 import
drwxrwxrwx 1 user_99    users      104 Feb 15 13:16 originals
drwxrwxrwx 1 user_99    users       76 Feb 16 15:59 storage

Please let me know if I can share any other output for better analysis.

I have already done below steps without any change. Still seeing the log message and now face crops.

- photoprism purge
- photoprism convert

- photoprism faces reset -f
- photoprism faces index

Any help is much appreciated. Thanks in advance.

**Comment by monarc83** [comment]
Can someone tell me at which path PhotoPrism is trying to save the face crop files like e.g. "61b19474a1818a8e1c713a2c1ff6141587a8294a_320x320_crop_2841e60e7134.jpg", so I can check the permissions.

Thanks in advance.

**Reply by graciousgrey** [reply]
The crops are saved in storage/cache/thumbnails/ the sub folders equal the first numbers of the file hash in your example it should be `storage/cache/thumbnails/6/1/b/`

**Comment by monarc83** [comment, marked as answer]
Okay I have found the issue. For some reason user and group for all files and folders within storage/cache/ were root:root, same for the json folder. I have changed them to user_99:users and now it is working. Don't know where this issue came from, since I have only created the container with default settings. A colleague of mine did the same and he hasn't had this issue.

**Reply by lastzero** [reply]
Happens when Docker creates the folders as root before starting the containers.

**Reply by monarc83** [reply]
Okay, I'll keep this mind. Is there a way to avoid this?

**Reply by monarc83** [reply]
I though below settings are there to avoid this!?

![grafik](https://user-images.githubusercontent.com/36295048/154438741-9669bbe7-e6a8-4863-b7ed-80722a143713.png)

**Reply by lastzero** [reply]
You could create directories manually upfront, we could provide a script/installer that does this, or run the container as root...

**Reply by lastzero** [reply]
In fact, the entrypoint script should also run chown when you use the environment variables to set the UID... unless you disabled it?

**Reply by lastzero** [reply]
https://github.com/photoprism/photoprism/blob/develop/docker/photoprism/entrypoint.sh#L60

**Reply by monarc83** [reply]
Haven't changed anything, so I guess I did not disable anything?! Just entered my host paths for Data Dir, Storage Path...

**Reply by lastzero** [reply]
Maybe `PHOTOPRISM_DISABLE_CHOWN: "true"` is set by default in the Unraid template? Make sure it's set to false if you want automatic ownership changes. Now that it's working, though, it shouldn't be needed.

**Reply by monarc83** [reply]
PHOTOPRISM_DISABLE_CHOWN isn't visible in the Unraid template so I guess it is using the default one which is "false"?!? One of my colleagues, also using the same image and template for PhotoPrism under Unraid didn't encountered this issue. Strange....

**Reply by zhanglix** [reply]
I'm caught by the same problem. I think its du to the way I starting indexing  ```docker exec -it photoprism photoprism index``` which is run by root in docker container..

**Reply by dhruvinsh** [reply]
Yes, i got the same issue, i did execute photoprism index from the container and end up with permission issue.  PHOTOPRISM_DISABLE_CHOWN did not help at this point. I need to look further.
